docs/enhancements/replace_broken_clusters.md

```diff
@@ -0,0 +1,69 @@
+# Automatically detect and replace broken Clusters
+
+## Summary
+
+Hibernating clusters often fail to transition to running state, but are assigned to ClusterClaims today. We would want to 
+validate a cluster can be successfully resumed and is running as expected before assigning it to a claim.
+
+## Motivation
+
+As we move towards being more proactive with having clusters ready as [hot spares](https://github.com/openshift/hive/pull/1434),
+we could come across instances where stale hibernated clusters could fail to resume and still be assigned to ClusterClaims.
+There's a need to detect broken clusters and replace them automatically.
+
+### Goals
+
+Always strive to assign a running cluster to a claim
+
+### Non Goals
+
+Fixing clusters (or attached claims) once a running cluster has already been assigned.
+
+## Proposal
+
+### Stand-by clusters
+
+Currently, we consider all installed clusters which are not set to be deleted as Assignable clusters. 
+Introduce another bucket of stand by clusters, which are installed and are not set for deletion, but are hibernating.
+A cluster is only assignable if it is running. A hibernating cluster has to be resumed successfully to be added to the 
+assignable clusters bucket.
+Some of the factors that might inhibit successful resume of a cluster that has been hibernating for a while include  
+unapproved certificates and expired credentials. Introduction of a separate bucket for hibernating clusters 
+allows for a validation before it is assigned to a claim.
+
+### Transition to Assignable clusters
+
+If a claim is created and there are no assignable clusters, we would resume a hibernating cluster and check if the resume
+was successful. The validating condition could be the expected powerstate matches the actual powerstate. If we encounter
+an error, we can mark that cluster to be deleted and try to transition the next available stand-by cluster.
```

Comment:
Can we expand on the workflow for how to decide when resume failed and how to handle that by deleting the cluster? deleting the cluster needs to be done in maxConcurrent budget etc.